rust/src/lib.rs

```diff
@@ -43,19 +43,28 @@ use treefile::treefile_read_impl;
 #[no_mangle]
 pub extern "C" fn rpmostree_rs_treefile_read(
     filename: *const libc::c_char,
+    arch: *const libc::c_char,
     fd: libc::c_int,
     error: *mut *mut glib_sys::GError,
 ) -> libc::c_int {
     // using an O_TMPFILE is an easy way to avoid ownership transfer issues w/ returning allocated
     // memory across the Rust/C boundary
 
+    let arch = if arch.is_null() {
+        None
+    } else {
+        let arch = unsafe { CStr::from_ptr(arch) };
+        Some(arch.to_string_lossy().into_owned())
+    };
+    let arch = arch.as_ref().map(String::as_str);
```

Comment:
I found writing what I expected to be an easy "convert a raw nullable string pointer to `Option<&str>`" to be more awkward than expected.  I was going to reach for `CStr::to_str().unwrap()` but that'd rely on our `panic=abort` setting, otherwise we'd try to unwind across FFI, which...OK it looks like [relatively recently](https://github.com/rust-lang/rust/pull/46833) turned into an abort.  So we probably can do that.

Reply:
OK, I reworked things here and I'm happier :arrow_down: - it's worth finding good patterns here as the FFI glue is going to be a risky area of the code.